plugin/arcconf/arcconf.py

```diff
@@ -398,22 +397,24 @@ def pools(self, search_key=None, search_value=None,
               flags=Client.FLAG_RSVD):
         lsm_pools = []
         getconfig_cntrls_info = self._get_detail_info_list()
-        sys_id = ''
         cntrl = 0
 
         for decoded_json in getconfig_cntrls_info:
             cntrl_num = str(int(cntrl + 1))
             sys_id = cntrl_num
             cntrl += 1
 
-            if 'LogicalDrive' in decoded_json['Controller']:
-                ld_infos = decoded_json['Controller']['LogicalDrive']
-                num_lds = len(ld_infos)
-                for ld in range(num_lds):
-                    ld_name = ld_infos[ld]['logicalDriveID']
-                    lsm_pools.append(
-                        Arcconf._arcconf_array_to_lsm_pool(ld_name, sys_id,
-                                                           cntrl_num))
+            if 'Array' in decoded_json['Controller']:
+                array_infos = decoded_json['Controller']['Array']
+                num_array = len(array_infos)
+                for array in range(num_array):
+                    array_id = array_infos[array]['arrayID']
+                    array_name = array_infos[array]['arrayName']
+                    block_size = array_infos[array]['blockSize']
+                    total_size = array_infos[array]['totalSize']
+                    unused_size = array_infos[array]['unUsedSpace']
+                    lsm_pools.append(Arcconf._arcconf_array_to_lsm_pool(
```

Comment:
Can you pass the `array` as single argument to `_arcconf_array_to_lsm_pool()`?

Like:
```python
def _arcconf_array_to_lsm_pool(arcconf_array):
    array_id = arcconf_array['arrayID']
    array_name = arcconf_array['arrayName']
    block_size = arcconf_array['blockSize']
    total_size = arcconf_array['totalSize']
    unused_size = arcconf_array['unUsedSpace']
    ...


lsm_pools.append(Arcconf._arcconf_array_to_lsm_pool(array_infos[array])

```

Reply:
Done - commit id - e5c44368c28265b40026398ae820e2b6ad6bcdb8